client/Scenes/Game_Main/Scripts/manager_connection.gd

```diff
@@ -5,63 +5,72 @@ var PORT = 6317
 var client = StreamPeerTCP.new()
 var username = "q"
 var password = "q"
-var list_characters = Array()
```

Comment:
where did this go?

Reply:
Just below.  The real change is that I don't instantiate list_characters when this script is loaded (this line becomes `var list_characters`).  This is important so that the list is recreated whenever the client receives a `character_list` response; if the client receives more than one `character_list` response, it will create a new list,  Without this change, it would continuously append characters to the `character_list` if you were to for instance login and log out over and over (not that this is currently possible).